### PR TITLE
Newarch dev edwin phase 2

### PR DESCRIPTION
**Pull request by edwin-huber**
Addresses:
1. #161 
1. #178 
1. #179 

Implemented List Flat Blobs  
Added tests for setting headers  
need to extend implementation and tests to cover the use of headers with any set / get request

### CODE REVIEW

File: src/blob/handlers/ContainerHandler.ts [hunk 1]
```diff
@@ -205,7 +205,67 @@ export default class ContainerHandler extends BaseHandler
...
+      options.maxresults,
+      marker
+    );
+
+    const blobItems: Models.BlobItem[] = [];
+    const blobPrefixes: Models.BlobPrefix[] = [];
+    const blobPrefixesSet = new Set<string>();
+    const prefixLength = options.prefix.length;
+
+    for (const blob of blobs) {
```
Comment (XiaoningLiu) [inline src/blob/handlers/ContainerHandler.ts]: `this.dataStore.listBlobs()` has filtered `prefix`, don't need to filter again here

File: src/blob/handlers/ContainerHandler.ts [hunk 2]
```diff
@@ -205,7 +205,67 @@ export default class ContainerHandler extends BaseHandler
...
+      }
+      const prefixPos = blob.name.indexOf(options.prefix);
+      // This is a blob with matching prefix
+      if (prefixPos >= 0) {
+        blob.deleted = blob.deleted !== true ? undefined : true;
+        blobItems.push(blob);
+      }
+    }
+
+    const iter = blobPrefixesSet.values();
```
Comment (XiaoningLiu) [inline src/blob/handlers/ContainerHandler.ts]: blobPrefixesSet doesn't any any values.

File: src/blob/handlers/ContainerHandler.ts [hunk 3]
```diff
@@ -205,7 +205,67 @@ export default class ContainerHandler extends BaseHandler
...
+      date: context.startTime,
+      serviceEndpoint,
+      containerName,
+      prefix: options.prefix,
+      marker: options.marker,
+      maxResults: options.maxresults || this.LIST_BLOBS_MAX_RESULTS_DEFAULT,
+      delimiter,
+      segment: {
+        blobItems,
+        blobPrefixes
```
Comment (XiaoningLiu) [inline src/blob/handlers/ContainerHandler.ts]: For `ContainerListBlobFlatSegmentResponse`, `blobPrefixes` is not needed. Can we put empty array to `blobPrefixes`?

File: src/blob/handlers/ContainerHandler.ts [hunk 4]
```diff
@@ -205,7 +205,67 @@ export default class ContainerHandler extends BaseHandler
     options: Models.ContainerListBlobFlatSegmentOptionalParams,
     context: Context
   ): Promise<Models.ContainerListBlobFlatSegmentResponse> {
-    throw new NotImplementedError(context.contextID);
+    const container = await this.getSimpleContainerFromStorage(context);
+    const request = context.request!;
```
Comment (XiaoningLiu) [inline src/blob/handlers/ContainerHandler.ts]: `request` variable seems useless

**Comment by XiaoningLiu** [comment]
@edwin-huber Thanks Edwin! That's really great! 

I Added some comments, please have a look!

File: src/blob/handlers/BlobHandler.ts [hunk 5]
```diff
@@ -125,11 +125,48 @@ export default class BlobHandler extends BaseHandler implements IBlobHandler {
     throw new NotImplementedError(context.contextID);
   }
 
+  // see also https://docs.microsoft.com/en-us/rest/api/storageservices/set-blob-properties
```
Comment (XiaoningLiu) [inline src/blob/handlers/BlobHandler.ts]: Can be improved to use standard JSDoc comments format.

You may want to install Visual Studio Code Plugin "Document This", which could help generate standard JSDoc comments for methods, classes, interfaces. After installing, goto previous line of code you want to comment, type "/**" then press "Enter", all comments generated!

Comment (XiaoningLiu) [reply src/blob/handlers/BlobHandler.ts]: Other Visual Studio Code plugins I'm using to help automatically code fix and format:
* tslint
* prettier
* TypeScript Import Sorter

Comment (edwin-huber) [reply src/blob/handlers/BlobHandler.ts]: Done

File: src/blob/handlers/ContainerHandler.ts (hunk 2 above)
Comment (edwin-huber) [reply src/blob/handlers/ContainerHandler.ts]: removed prefixes processing from flat list logic

File: src/blob/handlers/ContainerHandler.ts (hunk 3 above)
Comment (edwin-huber) [reply src/blob/handlers/ContainerHandler.ts]: removed prefixes

**Pull request merged**